src/control/README.md

```diff
@@ -1,113 +1,41 @@
+The DAOS Control Plane is written in Go. It is tasked with network
+and storage hardware provisioning and allocation in addition to
+instantiation and management of the DAOS I/O Server processes that
+run on the same host.
 
-The DAOS Server implements the [gRPC protocol](https://grpc.io/) to communicate with client gRPC applications and interacts with DAOS IO Servers through Unix domain sockets.
+## Design
 
-Multiple gRPC server modules are loaded by the control server. Currently included modules are security and management.
+The control directory contains a "cmd" subdirectory for server, agent
+and dmg applications. These applications import the client or server
+packages along with peripheral shared packages common, drpc, fault,
+logging and security where necessary to provide the given features.
 
-The Control Plane implements a replicated management service as part of the DAOS Server, responsible for handling distributed operations across the DAOS System.
+Specific library packages can be found in lib/ which provide access
+to native storage libraries through language bindings e.g. lib/spdk
+or specific formatting capabilities e.g. lib/hostlist or lib/txtfmt.
 
-The [management tool](cmd/dmg) is an example client application which can connect to both the [agent](cmd/agent) to perform security functions (such as providing credentials and retrieving security contexts) and to the local management server to perform management functions (such as storage device discovery).
+The pbin package provides framework for forwarding of requests to be
+executed by the privileged binary `daos_admin` on behalf of `daos_server`.
 
-## Documentation
+The provider package contains interface shims to the external environment,
+initially just to the Linux operating system.
 
-- [Management API](https://godoc.org/github.com/daos-stack/daos/src/control/client)
-- [Management internals](https://godoc.org/github.com/daos-stack/daos/src/control/server)
-- [Agent API](https://godoc.org/github.com/daos-stack/daos/src/control/client/agent)
-- [Agent internals](https://godoc.org/github.com/daos-stack/daos/src/control/security)
-- [dRPC](https://godoc.org/github.com/daos-stack/daos/src/control/drpc)
-- [server package](server/README.md)
-- [management tool package](cmd/dmg/README.md)
-- [client package](client/README.md)
-- [common package](common/README.md)
+The system package encapsulates the concept of the DAOS system and
+it's associated membership.
 
-## Architecture
+## Developer Documentation
 
-First a view of software component architecture:
+Please refer to package specific README's.
```

Comment:
I liked having the more specific links to the README files but it could get hard to maintain.